app/assets/javascripts/discourse/app/modifiers/observe-intersection.js

```diff
@@ -0,0 +1,18 @@
+import { modifier } from "ember-modifier";
+
+export default modifier((element, [callback], { threshold = 1 }) => {
+  const observer = new IntersectionObserver((entries) => {
+    entries.forEach(
+      (entry) => {
+        callback(entry);
+      },
+      { threshold }
+    );
+  });
```

Comment:
Could simplify to 

```suggestion
export default modifier((element, [callback], { threshold = 1 }) => {
  const observer = new IntersectionObserver((entries) => {
    entries.forEach((entry) => callback(entry), { threshold });
  });
```

I believe